workflow_scripts/check_model.py

```diff
@@ -1,16 +1,29 @@
 # SPDX-License-Identifier: Apache-2.0
 
+from cpuinfo import get_cpu_info
 import ort_test_dir_utils
 import onnxruntime
 import onnx
 import test_utils
 
 
+def has_vnni_support():
+    return 'avx512' in str(get_cpu_info()['flags'])
```

Comment:
I don't think this is strictly correct. From my reading there are CPUs that support avx512 "foundation" but not VNNI, which is an extension.
I think the flag to check for is `avx512_vnni` and maybe also `avx512_4vnniw`. 

Sources:

https://en.wikichip.org/wiki/x86/avx512_vnni
https://bugzilla.redhat.com/show_bug.cgi?id=1761678
https://unix.stackexchange.com/questions/43539/what-do-the-flags-in-proc-cpuinfo-mean

Reply:
Thank you for catching this! The function/PR name here might be misleading -- I made several experiments and concluded that machines with avx512f behave differently from machines without avx512f. Unfortunately I don't have machines with avx512_vnni support so I am not sure whether machines with avx512_vnni and machines with avx512f (and without avx512_vnni) behave the same. At least based on previous int8-related PRs, the output.pb produced by machines with avx512_vnni seems the same as the output.pb produced by machines with avx512f support only.

In summary, we need to figure out:
1. Machines without any avx512 support
2. Machines with avx512f support but without avx512_vnni support
3. Machines with avx512_vnni support

I am sure that 1 != 2 and 1 != 3, but I am not sure whether 2 == 3.

cc @mengniwang95 who is the main contributor of int8 models in ONNX Model Zoo. Feel free to comment if you have any insight. Thank you for the help!.